### PR TITLE
[NEW]  Automatic channels

### PR DESCRIPTION
**Pull request by karan-96**
continuing PR #7137

### CODE REVIEW

File: packages/rocketchat-plugin-handler/server/settings.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+RocketChat.settings.addGroup('plugin_settings', function() {
+	this.section('language', function() {
```
Comment (rodrigok) [inline packages/rocketchat-plugin-handler/server/settings.js]: Each plugin should add their own settings.

File: packages/rocketchat-plugin-handler/server/settings.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+RocketChat.settings.addGroup('plugin_settings', function() {
```
Comment (rodrigok) [inline packages/rocketchat-plugin-handler/server/settings.js]: Please, use a better name for the setting group, like `Automatic_Channels`

**Comment by karan-96** [comment]
@rodrigok @mitar, please review.

File: package.json [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (rodrigok) [inline package.json]: Do not introduce unnecessary changes

File: package.json [hunk 4]
```diff
@@ -85,18 +86,20 @@
...
-		"conventional-changelog-cli": "^1.3.2",
-		"eslint": "^4.5.0",
-		"postcss-cssnext": "^2.11.0",
-		"postcss-smart-import": "^0.7.5",
-		"simple-git": "^1.75.0",
-		"stylelint": "^8.0.0",
-		"stylelint-order": "^0.6.0",
-		"supertest": "^3.0.0"
-	},
+    "accept-language-parser": "^1.4.1",
```
Comment (rodrigok) [inline package.json]: Fix this indentation

File: packages/rocketchat-automatic-channels-geoip-category/server.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+		}
+	});
+	if (!countryNname) {
+		countryNname = null;
+	}
+
+	return countryNname;
+};
+
+
```
Comment (rodrigok) [inline packages/rocketchat-automatic-channels-geoip-category/server.js]: Remove this extra lines

File: packages/rocketchat-automatic-channels-handler/package.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+Package.onUse(function(api) {
+	api.use('accounts-base');
+	api.use('ecmascript');
+	api.use('rocketchat:lib');
+
+	api.mainModule('server/server.js', 'server');
+	api.addFiles('server/settings.js', 'server');
+});
+
+
```
Comment (rodrigok) [inline packages/rocketchat-automatic-channels-handler/package.js]: Remove this extra line

File: packages/rocketchat-automatic-channels-handler/server/server.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	});
+};
+
+Accounts.onLogin(function(user) {
+	if (!user.user._id || !user.user.username) {
+		return;
+	}
+	automaticChannelsHandler.categories.forEach((arrayItem) => {
+		const channelName = arrayItem.getChannelName(user);
+		if (channelName !== null && RocketChat.settings.get(arrayItem.enable)) {
```
Comment (rodrigok) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: Check plugin enabled before use it

File: packages/rocketchat-automatic-channels-handler/server/server.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+				}
+			} else {
+				// if room does not exist, create one
+				RocketChat.createRoom('c', channelName, 'rocket.cat', [user.user.username], false, {automatic: true, categoryName: arrayItem.categoryName});
+			}
+		}
+	});
+	// remove user from previously added automatic channels
+	removeUserFromOtherAutomaticChannels(user, automaticChannelsHandler.categories);
```
Comment (rodrigok) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: Why are you removing the user from channels after adding?

Comment (karan-96) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: user is being removed from previously added channels. 
For eg- If browser's language has been changed from french to english. Then user will be added to english and then removed from french.

Comment (rodrigok) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: Will this remove the user from the the channel he was just added in some case?

Comment (karan-96) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: I dont think so. There is a check for it here : ``` if (!channelNames.includes(arrayItem.name) ```

File: packages/rocketchat-automatic-channels-handler/server/server.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
+export const automaticChannelsHandler ={};
+automaticChannelsHandler.categories = [];
+
+function removeUserFromOtherAutomaticChannels(user, categories) {
+	const channelNames = categories.map(function(x) { return x.getChannelName(user); });
+
+	const userSubscriptions = RocketChat.models.Subscriptions.findByTypeAndUserId('c', user.user._id).fetch();
```
Comment (rodrigok) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: Can you check if there was returned channelNames (if plugins are enabled) before get all subscriptions? This is heavy

Comment (karan-96) [inline packages/rocketchat-automatic-channels-handler/server/server.js]: done

**Comment by rodrigok** [comment]
This feature is too specific, closing it in favor of any new Apps implementation.
